internal/controller/bucket/bucket.go

```diff
@@ -96,6 +96,7 @@ func Setup(mgr ctrl.Manager, o controller.Options, s *backendstore.BackendStore,
 		managed.WithLogger(o.Logger.WithValues("controller", name)),
 		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
 		managed.WithConnectionPublishers(cps...),
+		managed.WithCreationGracePeriod(creationGracePeriod),
```

Comment:
For me the goal of this period is not totally clear. Is it the timeout for `Create` function? If yes , `reconciliation-timeout` is lower so it would terminate the `Create` call earlier.

Reply:
My understanding is that creation grace period is much more simplistic and has a specific use case.

For more context, this is where it is set:
https://github.com/crossplane/crossplane-runtime/blob/54f79f799e0c5649b7a6c091a651937a36169797/pkg/reconciler/managed/reconciler.go#L577-L586

And this is the only place I can actually see it being used:
https://github.com/crossplane/crossplane-runtime/blob/54f79f799e0c5649b7a6c091a651937a36169797/pkg/reconciler/managed/reconciler.go#L938C2-L947

So in summary it is an arbitrary duration (30s by default) that crossplane must wait until after a managed resource has been created, before it can accept that it has been removed.

For example:
- Managed resource is created (with creation timestamp)
- Managed resource is deleted.
- Crossplane must wait until 30s (or whatever creation grace period is set to) after creation timestamp before Managed resource is removed - ie the Managed resource (bucket CR in our case) will hang until this time.

This is the only use I can find for this value, and 30s seems like a long time to wait :shrug: